pkg/expression/contextimpl/sessionctx.go

```diff
@@ -35,12 +38,25 @@ var _ exprctx.BuildContext = struct {
 
 // ExprCtxExtendedImpl extends the sessionctx.Context to implement `expression.BuildContext`
 type ExprCtxExtendedImpl struct {
-	sctx sessionctx.Context
+	sctx  sessionctx.Context
+	props contextopt.OptionalEvalPropProviders
 }
 
 // NewExprExtendedImpl creates a new ExprCtxExtendedImpl.
 func NewExprExtendedImpl(sctx sessionctx.Context) *ExprCtxExtendedImpl {
-	return &ExprCtxExtendedImpl{sctx: sctx}
+	impl := &ExprCtxExtendedImpl{sctx: sctx}
+	// set all optional properties
+	impl.setOptionalProp(currentUserProp(sctx))
+	// When EvalContext is created from a session, it should contain all the optional properties.
```

Comment:
"valContext is created from a session" means the current scene is a session from a user connection. At this time, we should provide all properties because all functions should work